src/coreclr/jit/lsra.cpp

```diff
@@ -12457,6 +12457,14 @@ regMaskTP LinearScan::RegisterSelection::select(Interval*    currentInterval,
         regMaskTP busyRegs = linearScan->regsBusyUntilKill | linearScan->regsInUseThisLocation;
         candidates &= ~busyRegs;
 
+#ifdef TARGET_ARM
+        // For TYP_DOUBLE on ARM, we can only use register for which the odd half is also available.
+        if (currentInterval->registerType == TYP_DOUBLE)
```

Comment:
If `busyRegs` does not contain both registers of the pairs, a more appropriate fix would be to find place where we miss setting 1 of the register and fix it. When we set `regMask` in `regsBusyUntilKill` or `regsInUseThisLocation`, we mostly use `getRegMask` which takes into account the "double register" as well. 
https://github.com/dotnet/runtime/blob/740ecd0a8db3f6a7714f3e15bac0774f033720f0/src/coreclr/jit/lsra.h#L1740-L1746

Reply:
> `genRegMask` versus `getRegMask`

I think we should use `getRegMask(reg, var_type)` version and delete `genRegMask(reg, var_type)`. They are doing same thing.

> So, say f2 is a candidate, and f3 is a busy register. we need to remove f2 from the candidate set due to f3 being busy.

Correct, thats what I expect, but it should happen at place that added `f3` as busy register. Do you know which of `regsBusyUntilKill` or `regsInUseThisLocation` does it gets marked as busy? The change looks correct, but I want to make sure we do not accidently mark a register (probably in jitstressregs) not part of current even/odd pair as busy.

Reply:
The case was `regsBusyUntilKill`.

> but it should happen at place that added f3 as busy register

If `f3` is marked as busy because it is a float (single-precision) type, then you wouldn't want to mark `f2` busy at that point. It's only when we're looking for candidates for a double that we need to mask off both `f2` and `f3` if `f3` is already marked as busy.

Reply:
Do you mind sharing the repro or the JitDump file?

Reply:
Sent a link to JitDump offline.

You can also repro locally using:

```
py src\coreclr\scripts\superpmi.py replay -f coreclr_tests -arch x86 -target_arch arm -target_os linux -jitoption JitOptRepeat#* -jitoption JitOptRepeatCount#2 -jitoption JitStressRegs#3
```

And looking for:

```
Assertion failed '!isRegBusy(physRegRecord->regNum, current->registerType)'
```

(for me, MC#459399)

(I’ve fixed the other asserts in https://github.com/dotnet/runtime/pull/94250)

Reply:
I tried to repro and see how `f3` ends up in `regsBusyUntilKill`. I am ok with this fix, but can we add `assert((busyRegs & 0x555555550000) == RBM_NONE)` which will confirm at least that we never have `f2` in this mask and hence we don't accidently remove `f1` from the mask?